backend-es/test/snapshots-out/Snapshot.EsPrecedence01.js

```diff
@@ -1,5 +1,5 @@
 const test1 = f => a => b => {
-  if ((f(a) ? f(b) : false) ? f(a) : true) { return f(a); }
+  if (f(a) && f(b) ? f(a) : true) { return f(a); }
```

Comment:
On the diff alone, I'm not sure what this is doing, but this is correct. However, here's another place where things could be optimized.

Reply:
Again, this is duplicated in the original code. We aren't doing any sort of common subexpression elimination.

Reply:
But I suppose if we are doing syntactic equality now, that's somewhat moot.

Reply:
I think something like this could be solved by #75, but it would need to be generalized to support arbitrary expressions, which should probably involve a hash of some kind. In this branch, we could say that `f(a)` refines to `true`, and replace all instances of `f(a)` with `true`.